README.md

```diff
@@ -512,6 +518,27 @@ The `React\Promise\PromisorInterface` provides a common interface for objects
 that provide a promise. `React\Promise\Deferred` implements it, but since it
 is part of the public API anyone can implement it.
 
+### AsyncInterop\Promise compatibility
+
+The [`PromiseInterface`](#promiseinterface) extends the
+[`AsyncInterop\Promise`](https://github.com/async-interop/promise) interface and
+thereby all promise implementations from React/Promise implement its `when()`
+method.
+
+The `AsyncInterop\Promise` interface is intended for interoperability with other
+libraries and their combinator and conversion functions.
+
+Note, that although the interface is named *Promise*, it has nothing to do with
+promises commonly known from the
+[CommonJS Promises/A](http://wiki.commonjs.org/wiki/Promises/A),
+[Promises/A+](https://promisesaplus.com) or
+[ECMAScript 2015 (ES6)](http://www.ecma-international.org/ecma-262/6.0/#sec-promise-constructor)
+specifications.
+
+If you're using React/Promise as your primary value placeholder implementation, 
+it is recommended to **not** use the `when()` method directly but the methods
+provided by the [`PromiseInterface`](#promiseinterface).
```

Comment:
@jsor You're right about `always`, thought it would only allow clean ups, missed that it supports forwarding as well.

My main point is only about this statement from your README:

> Calling done() transfers all responsibility for errors to your code. If an error (either a thrown exception or returned rejection) escapes the $onFulfilled or $onRejected callbacks you provide to done, it will be rethrown in an uncatchable way causing a fatal error.

Because in fact, those errors aren't uncatchable, they bubble up to `$deferred->resolve()` and then further up from there. To prevent exactly the bubbling up to `$deferred->resolve()`, the interop spec introduced the `ErrorHandler` which can either log such an error gracefully or abort the script if not such handler is defined.